utils/fetchEarthquakes.ts

```diff
@@ -154,8 +99,95 @@ export const fetchTodaySignificantEventsWithTimes = async (): Promise<
     }
   );
 
-  return res.data.features.map((feature: any) => ({
-    time: feature.properties.time,
-    magnitude: feature.properties.mag,
-  }));
+  const dailyEvents = res.data.features;
+
+  //? Is this too much to do here? Should atoms handle it?
+  // or is passing them data prepared straight up on the server a good move?
+
+  // get the total number of significant magnitude events
+  const dailyEventsTotal = dailyEvents.length;
+  // sort by time and magnitude - for chart
+  const dailyEventsWithTimes = dailyEvents.map(
+    (feature: EarthquakeFeature) => ({
+      time: feature.properties?.time,
+      magnitude: feature.properties?.mag,
+    })
+  );
+  // get the top 3 magnitude events
+  const topMagnitudeDailyEvents = dailyEvents
+    .sort(
+      (a: EarthquakeFeature, b: EarthquakeFeature) =>
+        b.properties?.mag - a.properties?.mag
+    )
+    .slice(0, 3);
+
+  return {
+    dailyEvents,
+    dailyEventsTotal,
+    dailyEventsWithTimes,
+    topMagnitudeDailyEvents,
+  };
```

Comment:
I can make this a little more OOP and add types so we always know what we're getting from this route - but that could be unecessary complexity (especially if we move these data transforms to atoms, but I'm still not sure on that route.)